main.go

```diff
@@ -109,9 +112,11 @@ func main() {
 		Namespace: managementClusterNamespace,
 	}
 	client := k8sclient.NewCluster(mgr.GetClient(), managementCluster)
+	session := session.Must(session.NewSession())
 
 	ec2Service := aws.NewEC2Client(ctx, client, managementCluster)
 	snsService := aws.NewSNSClient(ctx, snsTopic, client, managementCluster)
+	ramService := aws.NewRAMClient(ram.New(session, awsSdk.NewConfig()))
```

Comment:
I guess this not the correct way to init ram client. Would it be okay to read aws role identity inside main?

Reply:
Cant, we do it the same way we do it for EC2 and SNS service?
https://github.com/giantswarm/aws-network-topology-operator/blob/main/pkg/aws/ec2Client.go#L25-L62

Reply:
Do you mean changing the interface or implementing init logic inside main?

Reply:
Don't we need to use the identity referenced by the `AWSCluster`?

Reply:
But this is initialization for reconciler so there is no AWSCluster to work on.

Reply:
I would follow the same approach as in the other clients where we fetch the role for the management cluster as the RAm resource needs to be done in the same AWS account where the TGW exists that is the MC account

Reply:
Just in case it's useful to you, this is how we create the RAM client [in the `aws-resolver-rules-operator`](https://github.com/giantswarm/aws-resolver-rules-operator/blob/main/pkg/resolver/resolver.go#L247)